q2studio/api/workspace.py

```diff
@@ -78,9 +78,13 @@ def get_artifacts():
 @fail_gracefully
 def create_artifact():
     request_body = request.get_json()
+    try:
+        source_format = request_body['source_format']
+    except KeyError:
+        source_format = None
```

Comment:
I couldn't quite pin down why source_format was not ending up in the request body when a source format was not selected from the import dialogue.

It looks like this was not doing the trick: https://github.com/qiime2/q2studio/blob/015f858768aeadb7cb0b269ce46fe3a73649e045/app/js/containers/ArtifactGenerator.js#L29-L30

And also maybe something went wrong here:
https://github.com/qiime2/q2studio/blob/015f858768aeadb7cb0b269ce46fe3a73649e045/app/js/reducers/supertypes.js#L56-L62

Reply:
Please see my latest round of review for a suggestion on where to look.